azkaban-common/src/main/java/azkaban/executor/ExecutorManager.java

```diff
@@ -71,8 +80,20 @@
   private ConcurrentHashMap<Integer, ExecutableFlow> recentlyFinished =
       new ConcurrentHashMap<Integer, ExecutableFlow>();
 
-  private ExecutingManagerUpdaterThread executingManager;
+  /* map to easily access queued flows */
+  final private ConcurrentHashMap<Integer, Pair<ExecutionReference, ExecutableFlow>> queuedFlowMap =
```

Comment:
I am actually thinking since these code are mostly newly added, is this actually a good time to create a separate class and put them outside of this already big ExecutorManager?